Jenkinsfile

```diff
@@ -0,0 +1,48 @@
+@Library('jenkins-joylib@v1.0.0') _
+
+pipeline {
+
+    agent {
+        label joyCommonLabels(image_ver: '18.4.0')
+    }
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr: '30'))
+        timestamps()
+    }
+
+    stages {
+        stage('check') {
+            steps{
+                sh('make check')
+            }
+        }
+        stage('test') {
+            steps{
+                sh('make all test-unit')
```

Comment:
Hrm, doing `make all` then `re-clean` later, then the build which will have to `make all` again strikes as possibly heavy. I guess the npm cache might help us avoid having to re-download all npm packages twice? I'm not sure if we wipe out that case on 're-clean' here.

Reply:
I opened TOOLS-2349 to track the "re-clean is lame" issue.  I erred on the side of paranoid safety to start.